Timer: Added the initial requirements for the timer.

PawBud opened the pull request:
[Associated Discussion](https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/Adding.20an.20On-Air.20Timer)

### About the PR

- This PR aims to introduce a manual timer on the toolbar which will simply be a 2 state button. 

- The button will have a timer, on right-click, a menu will be open which will house the 3 buttons:- _Start, Stop & Reset_.

- This will give users an intuitive look at the timer.

#### How will the timer work during broadcasting?
The right-click will be disabled in such case, the timer will be automatically enabled when the connection is established and stop when the connection is ended. The in-depth working of this scenario is yet to be discussed.

Current State of the Toolbar:-
![Screenshot from 2021-04-01 21-28-08](https://user-images.githubusercontent.com/48093317/113321536-50033480-9331-11eb-86ad-884cc0dcc579.png)

**How am I approaching the problem?**
I am currently trying to put a QHBoxLayout that will house the 3 push buttons.
Next, I'll try to implement the logic for the Start, Stop, Reset button and the manual timer.

**Note**:- _The working of the timer is subject to change. This is a WIP feature and major changes must be expected as the feature is integrated._

PawBud committed:
Timer: Added the initial requirements for the timer.

This commit intrdouces an unactionable button in the toolbar
labelled timer. It is expected to support a right click
which will trigger a menu and host a stopwatch timer.

Note:- This commit is a WIP and doesn't function properly.
This commit will be referenced in future commits to mark
the completion of functionalities.

Holzhaus commented on src/widget/wtimerpopup.h:
```diff
@@ -0,0 +1,21 @@
+#include <QLineEdit>
+#include <QPushButton>
+
+class WCueMenuPopup : public QWidget {
```
This was intended to say WTimerPopup here and everywhere else in this file, right?

PawBud replied:
Yes, Thanks for pointing it out.

ronso0 replied:
> I am currently trying to put a QHBoxLayout that will house the 3 push buttons.

why not populate a regular QMenu with QActions? much less hazzle, see WCoverArtMenu
also that'll be a no-brainer to integrate with current menu styles.

PawBud replied:
I'll surely have a look thanks a lot! Not very well versed with the QT API

PawBud committed:
TimerMenu: Implements a Timer Menu class.

This commit introduces a WTimerMenu with start, stop and
reset actions. Currently, the logic for all the three
actions has not been implemented.

ronso0 commented:
@PawBud 
What about having these menu actions:
* Start
* Pause
* Restart (shortcut for Reset + Start)
* Reset
* `00:00` countdown duration in h:mm:ss format

PawBud commented:
> @PawBud
> What about having these menu actions:
> 
> * Start
> * Pause
> * Restart (shortcut for Reset + Start)
> * Reset
> * `00:00` countdown duration in h:mm:ss format
So I don't see the point of having a Restart button. I think it would create more ambiguity amongst our users.
As for the timer, once the user clicks on start (or reset if we decide to implement it), The label of the menu should switch to the timer in the specified format. Not sure how I am going to achieve that but I think I will find something the more I read the API docs.

PawBud commented:
> > @PawBud
> > What about having these menu actions:
> > 
> > * Start
> > * Pause
> > * Restart (shortcut for Reset + Start)
> > * Reset
> > * `00:00` countdown duration in h:mm:ss format
> 
> So I don't see the point of having a Restart button. I think it would create more ambiguity amongst our users.
> As for the timer, once the user clicks on start (or reset if we decide to implement it), The label of the menu should switch to the timer in the specified format. Not sure how I am going to achieve that but I think I will find something the more I read the API docs.

@ronso0 your thoughts on this?

ronso0 commented:
Honestly, this widget is has a very low prio for me currently, and I'm too lazy to think about fictional examples.
But I'd appreciate to have a somewhat working timer and test it, even though it may not be connected to any broadcasting or recording first.
Soo: Sure, let's your proposed menu first! After that we can discuss which controls we need at hand (always visible buttons) and which can go to the right-(left?)click menu, okay?
